hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupSystemTable.java

```diff
@@ -293,7 +293,19 @@ public void updateBackupInfo(BackupInfo info) throws IOException {
     }
     try (Table table = connection.getTable(tableName)) {
       Put put = createPutForBackupInfo(info);
-      table.put(put);
+      try {
+        table.put(put);
+      } catch (Exception e) {
+        // If the BackupInfo update can't be processed, then we should fall back to
+        // the previous BackupInfo, but also update it to reflect the failure.
+        LOG.error("Failed to update BackupInfo for {}. Marking as failed", info.getBackupId(), e);
```

Comment:
nit: why log and re-throw?

Reply:
To clarify that, since we lost this update to the BackupInfo, we're going to try to update its existing marker to failed